server/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -101,9 +104,14 @@ public ChecksumBlobStoreFormat(String codec, String blobNameFormat, CheckedFunct
      * @param name          name to be translated into
      * @return parsed blob object
      */
-    public T read(BlobContainer blobContainer, String name, NamedXContentRegistry namedXContentRegistry) throws IOException {
+    public T read(BlobContainer blobContainer, String name, NamedXContentRegistry namedXContentRegistry,
+                  BigArrays bigArrays) throws IOException {
         String blobName = blobName(name);
-        return deserialize(blobName, namedXContentRegistry, Streams.readFully(blobContainer.readBlob(blobName)));
+        try (ReleasableBytesStreamOutput out = new ReleasableBytesStreamOutput(bigArrays);
+             InputStream in = blobContainer.readBlob(blobName)) {
+            Streams.copy(in, out, false);
```

Comment:
```suggestion
            InputStream in = blobContainer.readBlob(blobName)) {
            Streams.copy(in, out, false);
```

Reply:
But `InputStream in` is part of the `try-with-resources` so it should align with the `ReleasableBytesStreamOutput out` above it shouldn't it?

Reply:
Raaah yes, I misread this line :( Sorry for the noise.